### ISSUE TITLE
Source code?

### ISSUE THREAD

**Issue by SachaTending** [Engine-Simulator/engine-sim-community-edition#45]
Is there any plans for releasing src?

**Comment by Abdullah-Ajeebi** [comment]
Found at https://github.com/ange-yaghi/engine-sim but its a bit old.

في الأحد، ٨ أكتوبر ٢٠٢٣, ٤:٥٦ م SachaTending ***@***.***> كتب:

**Comment by Kilgarragh** [comment]
Engine simulators old versions were open source. The fact that its closed now tells me that its fully intentional with no plans to release newer versions in source

**Comment by ange-yaghi** [comment]
@Kilgarragh Yes, there is nothing unintentional about the closed-source nature of recent Engine Simulator updates. As I've made clear to the community via Discord, I do not intend to update the open-source version of Engine Simulator. There were too many instances of abuse, companies taking credit for my work (as in, trying to pass off Engine Simulator as their own work) and various other negative side-effects of the release the codebase.

This is not to say that the code will never be released. At this point I see no incentive for me to release it and most requests to release it are "less than polite" and mostly angry demands that I do so without any reasoning for the action (besides the fact that they want free code). As far as I'm concerned, the core of Engine Simulator has been released with a very permissive license. It is sufficient as a reference implementation for anyone wanting to implement their own engine simulation/audio generator with no strings attached and no legal barriers. In my opinion, this is more than enough and until I'm convinced otherwise or start seeing much more respectful discourse on the matter, I will not be releasing any further source code related to this repository.

**Comment by raspitakesovertheworld** [comment]
I develop opensource code myself and know that there are some freeloaders that never wrote code and don't know how much work it is and they don't even value the contribution, so I wanted to say thank you for the code you did release as opensource, in the name of everybody that does appreciate your contribution!

**Comment by cryham** [comment]
> [@Kilgarragh](https://github.com/Kilgarragh) Yes, there is nothing unintentional about the closed-source nature of recent Engine Simulator updates. As I've made clear to the community via Discord, I do not intend to update the open-source version of Engine Simulator. There were too many instances of abuse, companies taking credit for my work (as in, trying to pass off Engine Simulator as their own work) and various other negative side-effects of the release the codebase.
> 
> This is not to say that the code will never be released. At this point I see no incentive for me to release it and most requests to release it are "less than polite" and mostly angry demands that I do so without any reasoning for the action (besides the fact that they want free code). As far as I'm concerned, the core of Engine Simulator has been released with a very permissive license. It is sufficient as a reference implementation for anyone wanting to implement their own engine simulation/audio generator with no strings attached and no legal barriers. In my opinion, this is more than enough and until I'm convinced otherwise or start seeing much more respectful discourse on the matter, I will not be releasing any further source code related to this repository.

Well that's sad that this happens. But I'm not too surprised.
You made an awesome software which got everyone interested, including their parents.
You did use MIT license, not GPL. But even big GPL software has those issues too.
Here is more about that:
https://en.wikipedia.org/wiki/GNU_General_Public_License#Legal_status
https://en.wikipedia.org/wiki/Gpl-violations.org
https://www.gpl-violations.org/news/

IMO, releasing sources, is not to avoid that happening, because it's impossible. Knowing how stupid commercial companies can be (e.g [video](https://www.youtube.com/watch?v=lEISWQubf5Y), and [whole channel](https://www.youtube.com/@moreperfectunion/videos), [also here](https://stallman.org/apple.html), etc), and trolls/people not understanding licenses, it just will happen.
I release under GPLv3 (e.g. [Stunt Rally 3](https://github.com/stuntrally/stuntrally3)) because I believe [Freedom](https://www.gnu.org/philosophy/free-sw.html) is most important for people. Even if I were one in a million doing so. Thus I despise everything commercial (especially biggest companies) and closed source. When you give people freedom, you can get more freedom in return: others contributing, creating, etc. Those who close sources and go comercial, are simply supporting modern slavery. Anyway thieves will exist.
Either way respect, and I wish you all the best.
